aspnetcore/fundamentals/environments.md

```diff
@@ -282,15 +300,167 @@ For Linux distros, use the `export` command at a command prompt for session-base
-* environment variables (set on each system where the app is hosted). See [Configuration: File configuration provider](xref:fundamentals/configuration/index#file-configuration-provider) and [Safe storage of app secrets in development: Environment variables](xref:security/app-secrets#environment-variables).
+::: moniker range=">= aspnetcore-3.0"
+
+* *appsettings* files (*appsettings.{Environment}.json*). See <xref:fundamentals/configuration/index#json-configuration-provider>.
+* Environment variables (set on each system where the app is hosted). See <xref:fundamentals/host/generic-host#environmentname> and <xref:security/app-secrets#environment-variables>.
+* Secret Manager (in the Development environment only). See <xref:security/app-secrets>.
+
+In addition to the recommended approaches, an app can also call <xref:Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.UseEnvironment*> when building the host. See <xref:fundamentals/host/generic-host#environmentname>.
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-3.0"
+
+* *appsettings* files (*appsettings.{Environment}.json*). See <xref:fundamentals/configuration/index#json-configuration-provider>.
+* Environment variables (set on each system where the app is hosted). See <xref:fundamentals/host/web-host#environment> and <xref:security/app-secrets#environment-variables>.
 * Secret Manager (in the Development environment only). See <xref:security/app-secrets>.
 
+In addition to the recommended approaches, an app can also call <xref:Microsoft.AspNetCore.Hosting.HostingAbstractionsWebHostBuilderExtensions.UseEnvironment*> when building the host. See <xref:fundamentals/host/web-host#environment>.
+
+::: moniker-end
+
+When the host is built, the last environment setting read by the app determines the app's environment. The app's environment can't be changed while the app is running.
+
 ## Environment-based Startup class and methods
 
+::: moniker range=">= aspnetcore-3.0"
+
+### Inject IWebHostEnvironment into Startup.Configure
+
+Inject <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> into `Startup.Configure`. This approach is useful when the app only requires adjusting `Startup.Configure` for a few environments with minimal code differences per environment.
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+    if (env.IsDevelopment())
+    {
+        // Development environment code
+    }
+    else
+    {
+        // Code for all other environments
+    }
+}
+```
+
+### Inject IWebHostEnvironment into the Startup class
+
+Inject <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> into the `Startup` constructor and assign the service to a field for use throughout the `Startup` class. This approach is useful when the app requires configuring `Startup` for only a few environments with minimal code differences per environment.
```

Comment:
~and assign the service to a field for use throughout the `Startup` class. ~
No writing the code in english. Just show them the code.

Inject <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment> into the `Startup`  
That's all you need.